src/components/customSearch/AddFieldOrGroupButton.tsx

```diff
@@ -0,0 +1,31 @@
+import { Button } from '@mui/material'
+import { SearchParameter } from '../../types/SimpleSearch'
+import AddIcon from '@mui/icons-material/Add'
+
+interface AddFieldOrGroupButtonProps {
+  push: (obj: any) => void
```

Comment:
```suggestion
  onClick: (obj: any) => void
```
What does the push func do ? Can we give it a better name ? Also why do we need to resend the prop ?

Reply:
The push function is coming from Formik. That's the function that adds a new index to the array.